packages/@aws-cdk/aws-scheduler-alpha/lib/target.ts

```diff
@@ -0,0 +1,231 @@
+import { Annotations, Duration, Names, PhysicalName, Token, TokenComparison } from 'aws-cdk-lib';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import { CfnSchedule } from 'aws-cdk-lib/aws-scheduler';
+import * as sqs from 'aws-cdk-lib/aws-sqs';
+import { Construct, IConstruct } from 'constructs';
+import { ScheduleTargetInput } from './input';
+import { ISchedule } from './schedule';
+
+export interface IScheduleTarget {
+  bind(_schedule: ISchedule): ScheduleTargetConfig;
+}
+
+export interface ScheduleTargetBaseProps {
+  readonly role?: iam.IRole;
+  /**
+   * The SQS queue to be used as deadLetterQueue.
+   *
+   * The events not successfully delivered are automatically retried for a specified period of time,
+   * depending on the retry policy of the target.
+   * If an event is not delivered before all retry attempts are exhausted, it will be sent to the dead letter queue.
+   *
+   * @default - no dead-letter queue
+   */
+  readonly deadLetterQueue?: sqs.IQueue;
+  readonly input?: ScheduleTargetInput;
+  /**
+   * The maximum age of a request that Scheduler sends to a target for processing.
+   *
+   * Minimum value of 60.
+   * Maximum value of 86400.
+   *
+   * @default Duration.hours(24)
+   */
+  readonly maxEventAge?: Duration;
+  /**
+   * The maximum number of times to retry when the target returns an error.
+   *
+   * Minimum value of 0.
+   * Maximum value of 185.
+   *
+   * @default 185
+   */
+  readonly retryAttempts?: number;
+}
+
+/**
+ * Properties for an Schedule Target
+ */
+export interface ScheduleTargetConfig {
```

Comment:
While you removal makes the code a bit easier, the use when specifying paramaters for a specific target becomes a bit harder because codecompletion will show you a lot of options to choose from while only one of the parameter properties should be filled in any specific case.

I personally don't like it. I think it is fine for CDK internal code to be complicated if it enhanced the use by users of the library.

Reply:
Did you maybe confuse it  with `ScheduleTargetBaseProps`? The interface that the customers will use is `ScheduleTargetBaseProps`, while the interface in discussion - `ScheduleTargetConfig` is used as output of `bind` operation. I think the documentation is misleading and I should update it.

Reply:
I see you point now. I guess I just have to play around with it a bit :-)